docs/specs/authentication-flow.md

```diff
@@ -356,6 +357,104 @@ reauth_flows:
     - authentication: secondary_sms_code
 ```
 
+### Flow selection
+
+By default, authentication flow with name `default` is used. Since there can be multiple flows of same type, we want client to be able to select a flow group to use.
+
+For example, given following authentication flows:
+
+```yaml
+authentication_flow:
+  signup_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  signup_login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  reauth_flows:
+    - name: sms_code
+      steps:
+      # ...
+    - name: password
+      steps:
+      # ...
+  promote_flows:
+    - name: password
+      steps:
+      # ...
+    - name: sms_code
+      steps:
+      # ...
+  account_recovery_flows:
+    - name: email
+      steps:
+      # ...
+    - name: sms
+      steps:
+      # ...
+```
+
+Authentication flow groups need to be defined and whitelisted per client using the following configuration:
+
+```yaml
+ui:
+  authentication_flow:
+    groups:
+    - name: oauth_only
+      login_flow: oauth_only
+      signup_flow: oauth_only
+      reauth_flow: password
+      promote_flow: password
+      account_recovery_flow: email
+    - name: email_password_2fa
+      login_flow: email_password_2fa
+      signup_flow: email_password_2fa
+      reauth_flow: sms_code
+      promote_flow: sms_code
+      account_recovery_flow: sms
+
+oauth:
+  clients:
+    - client_id: public_app
+      # Only allow OAuth login for public_app
+      x_authentication_flow_group_allowlist:
+      - oauth_only
+    - client_id: internal
+      # Allow both OAuth login and email_password_2fa for internal
+      x_authentication_flow_group_allowlist:
```

Comment:
Mention this new client metadata in `docs/specs/oidc.md#custom-client-metadata`. The description is a backlink to a proper section in authentication-flow-in-default-ui.md